### PR TITLE
#166237757 initialize express server with environment variables

### PR DESCRIPTION
**Pull request by obayomi96**
#### What does this PR do?
This PR Initializes the express server with environment variables

#### Description of Task to be completed?
Dev script to start the project in development mode (using nodemon)
start script to start the project in production mode (using node)
Integrate dotenv for managing environment variables
ignore .env in .gitginore
Readme.md file updated with instructions on how to run the project
The WIldcard route to handle any endpoint, return a JSON message 'hello books deferral'

Have the following endpoints working
localhost:4000/api/v1

#### What are the relevant pivotal tracker stories?
#166237757

Screenshot (234)

### CODE REVIEW

**Comment by ayodejiAA** [comment]
Looks okay. Nice work.

**Comment by chialuka** [comment]
Looks good. Thank you for handling the README too

**Review by tolulope-od — COMMENTED** [review]
This is great stuff @obayomi96 I noticed to added a catch-all for all undefined routes. I think it wouldn't be a bad idea to add an error handling middleware also. Do you think you can add that to this branch? Again, good stuff. Thanks for kicking things off for us 👍

**Comment by obayomi96** [comment]
> This is great stuff @obayomi96 I noticed to added a catch-all for all undefined routes. I think it wouldn't be a bad idea to add an error handling middleware also. Do you think you can add that to this branch? Again, good stuff. Thanks for kicking things off for us 👍

Thanks @tolulope-od  It's a good idea to implement that as it didn't occur to me before.  I will have to do it at night. If other PRs have been merged I will do it in another branch as a chore. Thanks for pointing that out

**Comment by ayodejiAA** [comment]
I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?

**Comment by obayomi96** [comment]
> I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?

True @Xwebyna, I did that initially, and it really gave me a taugh time. Untill I had to leave it in the dependency. The app kept breaking. Maybe if someones moves it we'll see

**Comment by ayodejiAA** [comment]
> > I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?
> 
> True @Xwebyna, I did that initially, and it really gave me a taugh time. Untill I had to leave it in the dependency. The app kept breaking. Maybe if someones moves it we'll see

That's okay.  @obayomi96. One of us will look into it. Great work.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    res.status(404).json({
+      msg: 'Opps!!, the page you are looking for cannot be found',
+    });
+  });
+
+app.listen(port, () => {
+    console.log(`Server started on port ${port}`);
+});
+
+export default app;
```
Comment (akpante3) [inline index.js]: @obayomi96 nice work here, please create a new line here.

File: routes/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+    res.status(200).json({
+        message: "Hello Books Deferral"
+    });
+});
+
+export default router;
```
Comment (akpante3) [inline routes/index.js]: @obayomi96 new line here too

File: index.js (hunk 1 above)
Comment (obayomi96) [reply index.js]: Noted, will do that now.

File: routes/index.js (hunk 2 above)
Comment (obayomi96) [reply routes/index.js]: Noted

**Pull request merged**